### ISSUE TITLE
Compilation fails on MacOS

### ISSUE THREAD

**Issue by mtimofiiv** [briansmith/ring#791]
Hello!

On MacOS 14.1 and Rust 1.32 and nightly 1.34, `ring` does not build. I am trying to install `rocket`, and the dependency version is `0.13.5`, but having tried to also install `0.14.5`, I get the same results.

I have attached the log: [log.txt](https://github.com/briansmith/ring/files/2845461/log.txt)

As far as I know the nullability messages are warnings, so they should not by themselves ruin it.

Clang details:

```
Apple LLVM version 10.0.0 (clang-1000.10.44.4)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Xcode version: Version 10.1 (10B61)

**Comment by mtimofiiv** [comment]
Additionally, I tried to install the master branch of this module as well, since [the test appears to be passing](https://travis-ci.org/briansmith/ring/jobs/489842647). There is a different error, and it still refuses to build:

```
--- stderr
clang: error: argument unused during compilation: '-L/usr/local/lib' [-Werror,-Wunused-command-line-argument]
thread 'main' panicked at 'execution failed', /Users/fiiv/.cargo/git/checkouts/ring-51509f604a4fd25b/78b1877/build.rs:638:9
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by briansmith** [comment]
Thanks for the report.

Regarding the warnings in `<stdlib.h>`, the master branch doesn't include `<stdlib.h>` any more so that's why it's fixed, I guess. I released 0.14.6 with that change.

I suspect that `-L/usr/local/lib` is added by [cc-rs](https://github.com/alexcrichton/cc-rs), so maybe it is a bug in cc-rs?

Regardless, If you build another project that uses *ring* as a dependency, `-Werror` won't be used so *ring* should build fine.

**Comment by briansmith** [comment]
Another possibility is that the attribute is being added because of our `libc` crate dependency. On macOS we use libc only for its types (`size_t` and `c_int` in particular) and this is relatively new. I'm not sure how to tell Rust how to use `libc` but not link the C standard library.

**Comment by briansmith** [comment]
> On MacOS 14.1

I think this is maybe specific to macOS 14. Unfortunately as of today Travis CI doesn't support macOS 14 and it may be a while before I get macOS 14 on my own Mac. So, realistically, it would probably be faster for somebody else to debug and resolve this.

**Comment by andrewtj** [comment]
I've been unable to replicate this on 10.14.3. I have the same Xcode release but my clang version differs:
```
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
This [table](https://en.wikipedia.org/wiki/Xcode#Latest_versions) suggests `clang-1000.10.44.4` is from an earlier (beta?) release.

**Comment by mtimofiiv** [comment]
I initialised a fresh project, no other deps. I added `ring = "0.14.6"` under `[dependencies]`. Running `cargo run` actually caused no problems, a clean install. So I suspect this is to do with the older versions under this specific MacOS version. I found the same to be true with `0.14.5`. I did not update `clang` or anything so not sure why it's all of a sudden building fine.

`MacOS 14` did change a lot, like dropping 32-bit support, so as a result a lot of different pieces of software just won't even build anymore.

`ring = "0.13.5"` still crashes with errors however, but since it's an old version that's obviously no big deal.

Thanks @briansmith and @andrewtj, I'll go post in the `rocket` issue tracker to update their `ring` version!
